landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+
+    Define grid and initial topography
+
+    * 5x4 grid with baselevel in the lower left corner
+    * All other boundary nodes closed
+    * Initial topography is plane tilted up to the upper right with noise
+
+    >>> mg = RasterModelGrid((5, 4), xy_spacing=10.0)
+    >>> mg.set_status_at_node_on_edges(
+    ...     right=mg.BC_NODE_IS_CLOSED,
+    ...     top=mg.BC_NODE_IS_CLOSED,
+    ...     left=mg.BC_NODE_IS_CLOSED,
+    ...     bottom=mg.BC_NODE_IS_CLOSED,
+    ... )
+    >>> mg.status_at_node[1] = mg.BC_NODE_IS_FIXED_VALUE
+    >>> mg.add_zeros("topographic__elevation", at="node")
+    array([ 0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.])
+    >>> rand_noise=np.array(
+    ...     [
+    ...         0.00436992,  0.03225985,  0.03107455,  0.00461312,
+    ...         0.03771756,  0.02491226,  0.09613959,  0.07792969,
+    ...         0.08707156,  0.03080568,  0.01242658,  0.08827382,
+    ...         0.04475065,  0.07391732,  0.08221057,  0.02909259,
+    ...         0.03499337,  0.09423741,  0.01883171,  0.09967794,
+    ...     ]
+    ... )
+    >>> mg.at_node["topographic__elevation"] += (
+    ...     mg.node_y / 10. + mg.node_x / 10. + rand_noise
+    ... )
+    >>> U = 0.001
+    >>> dt = 100
+
+    Instantiate flow accumulation and lateral eroder and run each for one step
+
+    >>> fa = FlowAccumulator(
+    ...     mg,
+    ...     surface="topographic__elevation",
+    ...     flow_director="FlowDirectorD8",
+    ...     runoff_rate=None,
+    ...     depression_finder=None,
+    ... )
+    >>> latero = LateralEroder(mg, latero_mech="UC", Kv=0.001, Kl_ratio=1.5)
+
+    Run one step of flow accumulation and lateral erosion to get the dzlat array
+    needed for the next part of the test.
+
+    >>> fa.run_one_step()
+    >>> mg, dzlat = latero.run_one_step(dt)
+
+    Evolve the landscape until the first occurence of lateral erosion. Save arrays
+    volume of lateral erosion and topographic elevation before and after the first
+    occurence of lateral erosion
+
+    >>> while min(dzlat) == 0.0:
+    ...     oldlatvol = mg.at_node["volume__lateral_erosion"].copy()
+    ...     oldelev = mg.at_node["topographic__elevation"].copy()
+    ...     fa.run_one_step()
+    ...     mg, dzlat = latero.run_one_step(dt)
+    ...     newlatvol = mg.at_node["volume__lateral_erosion"]
+    ...     newelev = mg.at_node["topographic__elevation"]
+    ...     mg.at_node["topographic__elevation"][mg.core_nodes] += U * dt
+
+    Before lateral erosion occurs, *volume__lateral_erosion* has values at
+    nodes 6 and 10.
+
+    >>> np.around(oldlatvol, decimals=0)
+    array([  0.,   0.,   0.,   0.,
+             0.,   0.,  79.,   0.,
+             0.,   0.,  24.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.])
+
+
+    After lateral erosion occurs at node 6, *volume__lateral_erosion* is reset to 0
+
```

Comment:
@alangston these tests are a really nice introduction to how the component works. Thanks for making them!